examples/src/lib.rs

```diff
@@ -459,9 +468,20 @@ impl ExampleBase {
                 .bind_image_memory(depth_image, depth_image_memory, 0)
                 .expect("Unable to bind depth image memory");
 
+            let fence_create_info =
+                vk::FenceCreateInfo::builder().flags(vk::FenceCreateFlags::SIGNALED);
+
+            let draw_commands_reuse_fence = device
+                .create_fence(&fence_create_info, None)
+                .expect("Create fence failed.");
+            let setup_commands_reuse_fence = device
+                .create_fence(&fence_create_info, None)
+                .expect("Create fence failed.");
+
             record_submit_commandbuffer(
                 &device,
                 setup_command_buffer,
+                setup_commands_reuse_fence,
```

Comment:
Wait is that actually correct now? Because `record_submit_commandbuffer` only waits on the 2nd call now, and we do the setup only once.

Reply:
There's no need to wait for the setup command buffer on CPU side if nobody will reuse that command buffer.

Reply:
You don't even need a separate setup command buffer. That could be removed.